Packs/PcapAnalysis/Scripts/PcapFileExtractor/PcapFileExtractor.py

```diff
@@ -0,0 +1,195 @@
+
+    return excluded_files
+
+
+def upload_files(excluded_files, dir_path, file_path):
+    """
+
+    :param excluded_files: excluded files
+    :param dir_path: dir path for the files
+    :param file_path: the path to the pcap file
+    :return:
+    """
+    filenames = []  # type: ignore
+    # recursive call over the file system top down
+    for root, directories, files in os.walk(dir_path):
+        for f in files:
+            # skipping previously existing files
+            # adding it to the extracted pcap files list
+            if f not in excluded_files and isfile(os.path.join(root, f)):
+                filenames.append(os.path.join(root, f))
+
+    if len(filenames) == 0:
+        return_error('Could not find files')
+
+    else:
+        results = []
+        context = []
+        protocol, packet_data = find_files_protocol(file_path)
+
+        md5 = hashlib.md5()
+        sha1 = hashlib.sha1()
+        sha256 = hashlib.sha256()
+        files_base_names = [os.path.basename(file_path) for file_path in filenames]  # noqa[F812]
+        files_dic = {file_path: os.path.basename(file_path) for file_path in filenames}
+
+        for file_path, file_name in files_dic.items():
+            for data in packet_data:
+                packet_number = data.split()[0]
+                for packet_number in packet_data:
+                    data = [i for i in packet_number.split()]
+                    source_ip = data[2]
+                    dest_ip = data[4]
+
+            with open(file_path, 'rb') as _file:
+                demisto.results(fileResult(file_name, _file.read()))
+
+            with open(file_path, 'rb') as _file:
+                data = _file.read()
+                md5.update(data)
+                sha1.update(data)
+                sha256.update(data)
+
+            context.append({
+                'FileMD5': md5.hexdigest(),
+                'FileSHA1': sha1.hexdigest(),
+                'FileSHA256': sha256.hexdigest(),
+                'FileName': file_name,
+                'FileSize': os.path.getsize(file_path),
+                'DetectedInProtocol': protocol,
+                'FileExtension': os.path.splitext(file_name)[1],
+                'SourceIP': source_ip,
+                'DestinationIP': dest_ip
+            })
+
+            ec = {
+                'PcapExtractedFiles(val.FileMD5 === obj.FileMD5)': context
+            }
+
+        results.append(
+            {
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': {'extractedFiles': files_base_names},
+                'EntryContext': ec,
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': tableToMarkdown('Pcap Extracted Files',
+                                                 [{'name': file_name} for file_name in files_base_names])
+            })
+
+        demisto.results(results)
```

Comment:
lets remove this line and call `return_outputs` in the `main` func so it will be easier for unti testing